cmd/options/node_options.go

```diff
@@ -21,6 +21,15 @@ import (
 )
 
 // NodeOptions contains options and configuration settings for the node service.
-type NodeOptions struct{}
+type NodeOptions struct {
+	// VolumeAttachLimit specifies the value that shall be reported as "maximum number of attachable volumes"
+	// in CSINode objects. It is similar to https://kubernetes.io/docs/concepts/storage/storage-limits/#custom-limits
+	// which allowed administrators to specify custom volume limits. Note that this solution is just a "stop-gap"
+	// solution until a more sophisticated solution can be implemented (dynamically discovering the maximum number of
```

Comment:
hmm, I am not sure about documenting this as "stop-gap" where no other solution exists or being worked on. I would drop that wording and document it as such until an alternative presents itself and then we can deprecate this.

Reply:
Hm, but isn't it exactly like this? It's a stop-gap solution because the alternative would be to not have any possibility to migrate to CSI in case one depends on the Kubernetes feature allowing to overwrite the volume limits. Also, nobody is working on the "real" solution, or? The last update from the ticket is that the easiest implementation would require AWS to expose this information, e.g. via metadata service.

Anyways, what do you suggest to write here instead? Shall I add that the flag is planned to be deprecated in the future when a better solution is implemented?

Reply:
I think `KUBE_MAX_PD_VOLS` environment variable should work for CSI drivers too - can you open a github issue in kubernetes/kubernetes repo.

Reply:
@gnufied I've rephrased this part and also the section in `docs/design.md`.

> I think KUBE_MAX_PD_VOLS environment variable should work for CSI drivers too - can you open a github issue in kubernetes/kubernetes repo.

According to @hardikdr it doesn't work, but he is currently reproducing the issue an then will report it to k/k. But yes, independent of this it's still reasonable to have it configurable as you initially suggested in #347.

Reply:
The env-var `KUBE_MAX_PD_VOLS` doesn't have any effect on max volumes attachments, checked on k8s 1.18 with CSI drivers. But I guess, this should better be handled at the CSI-drivers than scheduler.